lua/nvim-treesitter/shell_command_selectors.lua

```diff
@@ -289,6 +289,31 @@ function M.select_download_commands(repo, project_name, cache_folder, revision,
     local git_folder = utils.join_path(cache_folder, project_name)
     local clone_error = "Error during download, please verify your internet connection"
 
+    -- Running `git clone` or `git checkout` while running under Git (such as
+    -- editing a `git commit` message) will likely fail to install parsers
+    -- (such as 'gitcommit') and can also corrupt the index file of the current
+    -- Git repository. Check for typical git environemnt variables and abort if found.
```

Comment:
I ran into this issue a couple weeks ago and made a note to look into it later, only to find this PR! Thank you for fixing this - getting a corrupt index file was quite surprising, so I'm glad that's resolved.

I noticed a minor typo in this comment which I've fixed in https://github.com/nvim-treesitter/nvim-treesitter/pull/6986